package/com.unity.formats.usd/Tests/Runtime/AnimatedUSDFixture.cs

```diff
@@ -0,0 +1,173 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NUnit.Framework;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Playables;
+using UnityEngine.TestTools;
+using UnityEngine.Timeline;
+using USD.NET;
+using Object = UnityEngine.Object;
+
+namespace Unity.Formats.USD.Tests
+{
+     class AnimatedUsdFixture
+    {
+        readonly List<string> m_filesToDelete = new List<string>();
+        UsdAsset m_usdRoot;
+        
+        [UnitySetUp]
+        public IEnumerator SetUp()
+        {
+            InitUsd.Initialize();
+            
+            var curve = AnimationCurve.Linear(0, 50, 1, 100);
+            var clip = new AnimationClip();
+            clip.SetCurve("", typeof(Transform), "localPosition.x", curve);
+            clip.SetCurve("", typeof(Camera), "field of view", curve);
+
+            CreateTimeline(out var director, out var timeline);
+            
+            var aTrack = timeline.CreateTrack<AnimationTrack>(null, "CubeAnimation");
+            aTrack.CreateClip(clip).displayName = "CamClip";
+            var cam = new GameObject("Camera").AddComponent<Camera>().gameObject;
+            director.SetGenericBinding(aTrack, cam.AddComponent<Animator>());
+            var usdRecordTrack = timeline.CreateTrack<UsdRecorderTrack>(null, "");
+            var usdRecorderClip = usdRecordTrack.CreateDefaultClip();
+            usdRecorderClip.start = 0;
+            usdRecorderClip.duration = 2;
+            var usdRecorderAsset = usdRecorderClip.asset as UsdRecorderClip;
+            usdRecorderAsset.m_usdFile = "Assets/" + Path.GetFileNameWithoutExtension(Path.GetTempFileName())+".usd";
+            m_filesToDelete.Add(usdRecorderAsset.m_usdFile);
+            usdRecorderAsset.m_exportRoot = new ExposedReference<GameObject> {exposedName = Guid.NewGuid().ToString()};
+            director.SetReferenceValue(usdRecorderAsset.m_exportRoot.exposedName, cam);
+            Time.captureFramerate = (int)timeline.editorSettings.fps;
+            director.Play();
+            while (director.time <= 1.1)
+                yield return null;
+            director.Stop();
+            yield return null;
+            var stage = pxr.UsdStage.Open(usdRecorderAsset.m_usdFile, pxr.UsdStage.InitialLoadSet.LoadAll);
+            var scene =  Scene.Open(stage);
+            m_usdRoot = ImportSceneAsGameObject(scene).GetComponent<UsdAsset>();
+            scene.Close();
+            m_usdRoot.name = "Potato";
+        }
+
+        [Test]
+        public void TestCamAnimation()
```

Comment:
Similarly, I think a short comment of what this is testing and how it's testing it would be helpful here too